instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigProperties.java

```diff
@@ -29,37 +31,87 @@ public class SpringConfigProperties implements ConfigProperties {
   private final ExpressionParser parser;
   private final OtlpExporterProperties otlpExporterProperties;
   private final OtelResourceProperties resourceProperties;
-  private final PropagationProperties propagationProperties;
+  private final SpringOtelProperties springOtelProperties;
   private final ConfigProperties otelSdkProperties;
+  private final ConfigProperties customizedResourceProviderProperties;
+
+  private static final Map<String, Function<SpringOtelProperties, List<String>>> LIST_PROPERTIES =
+      new HashMap<>();
+
+  static final String DISABLED_KEY = "otel.java.disabled.resource.providers";
+  static final String ENABLED_KEY = "otel.java.enabled.resource.providers";
+
+  static {
+    LIST_PROPERTIES.put(ENABLED_KEY, SpringOtelProperties::getJavaEnabledResourceProviders);
+    LIST_PROPERTIES.put(DISABLED_KEY, SpringOtelProperties::getJavaDisabledResourceProviders);
+    LIST_PROPERTIES.put(
+        "otel.experimental.metrics.view.config",
+        SpringOtelProperties::getExperimentalMetricsViewConfig);
+    LIST_PROPERTIES.put(
+        "otel.experimental.resource.disabled.keys",
+        SpringOtelProperties::getExperimentalResourceDisabledKeys);
+    LIST_PROPERTIES.put("otel.propagators", SpringOtelProperties::getPropagators);
+  }
 
   public SpringConfigProperties(
       Environment environment,
       ExpressionParser parser,
       OtlpExporterProperties otlpExporterProperties,
       OtelResourceProperties resourceProperties,
-      PropagationProperties propagationProperties,
+      SpringOtelProperties springOtelProperties,
       ConfigProperties otelSdkProperties) {
     this.environment = environment;
     this.parser = parser;
     this.otlpExporterProperties = otlpExporterProperties;
     this.resourceProperties = resourceProperties;
-    this.propagationProperties = propagationProperties;
+    this.springOtelProperties = springOtelProperties;
     this.otelSdkProperties = otelSdkProperties;
+    this.customizedResourceProviderProperties =
+        createCustomizedProperties(otelSdkProperties, springOtelProperties);
+  }
+
+  private static void addList(
+      Map<String, String> map, String key, List<String> list, ConfigProperties configProperties) {
+    if (!list.isEmpty()) {
+      map.put(key, String.join(",", list));
+    } else {
+      String defaultValue = configProperties.getString(key);
+      if (defaultValue != null) {
+        map.put(key, defaultValue);
```

Comment:
Could you please explain when the evaluation of a default value from `DefaultConfigProperties` is required.

Reply:
the variable names were confusing - changed to

```java
 if (!springList.isEmpty()) {
      return Collections.singletonMap(key, String.join(",", springList));
    } else {
      String otelList = configProperties.getString(key);
      if (otelList != null) {
        return Collections.singletonMap(key, otelList);
      }
    }
```